java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -301,35 +309,88 @@ public void indexingComplete(Set<URL> indexedRoots) {
             EditorCookie ec = file.getLookup().lookup(EditorCookie.class);
             Document doc = ec.openDocument();
             final int caret = Utils.getOffset(doc, params.getPosition());
-            ParserManager.parse(Collections.singletonList(Source.create(doc)), new UserTask() {
-                @Override
-                public void run(ResultIterator resultIterator) throws Exception {
-                    TokenSequence<JavaTokenId> ts = resultIterator.getSnapshot().getTokenHierarchy().tokenSequence(JavaTokenId.language());
-                    if (ts.move(caret) == 0 || !ts.moveNext()) {
-                        if (!ts.movePrevious()) {
-                            ts.moveNext();
+            String mimeType = file.getMIMEType();
+            if ("text/x-java".equals(mimeType)) {
+                ParserManager.parse(Collections.singletonList(Source.create(doc)), new UserTask() {
```

Comment:
Ideally this special `text/x-java` code could be moved to Java module and exposed as `CompletionCollector`.

Reply:
Done.